tools/serverpod_cli/lib/src/generator/code_generation_collector.dart

```diff
@@ -43,10 +43,26 @@ class CodeGenerationCollector extends CodeAnalysisCollector {
       return;
     }
 
-    log.error(
-      'Found ${errors.length} error${errors.length == 1 ? '' : 's'}.',
-      newParagraph: true,
-    );
+    var logMessage =
+        'Found ${errors.length} issue${errors.length == 1 ? '' : 's'}.';
+
+    var severityErrors = errors.whereType<SourceSpanSeverityException>();
```

Comment:
Question: Is there ever a case where we would collect SourceSpanException that have `error` severity but are not of `SourceSpanSeverityException` type that we miss here?

Reply:
Since the `SourceSpanException` does not have a severity we treat all of them like severity `error`. 

Perhaps the code is not straight forward here, but since we have an early return if there are no errors, it means that if severityErrors is empty we have `SourceSpanException`s in our list.

The row below this checks for this `severityErrors.isEmpty`. Hence we code this case here.